Loop/View Controllers/SettingsTableViewController.swift

```diff
@@ -464,29 +473,50 @@ final class SettingsTableViewController: UITableViewController {
+                        dataManager?.loopManager.settings.glucoseTargetRangeSchedule = newSchedule
+                        tableView.reloadRows(at: [indexPath], with: .automatic)
+                    }
+                )
 
-                scheduleVC.delegate = self
-                scheduleVC.title = NSLocalizedString("Correction Range", comment: "The title of the glucose target range schedule screen")
+                let hostingController = DismissibleHostingController(rootView: editor, onDisappear: {
+                    tableView.deselectRow(at: indexPath, animated: true)
+                })
 
-                if let schedule = dataManager.loopManager.settings.glucoseTargetRangeSchedule {
-                    var overrides: [TemporaryScheduleOverride.Context: DoubleRange] = [:]
-                    overrides[.preMeal] = dataManager.loopManager.settings.preMealTargetRange.filter { !$0.isZero }
-                    overrides[.legacyWorkout] = dataManager.loopManager.settings.legacyWorkoutTargetRange.filter { !$0.isZero }
-                    scheduleVC.setSchedule(schedule, withOverrideRanges: overrides)
-                }
+                present(hostingController, animated: true)
+            case .correctionRangeOverrides:
+                let unit = dataManager.loopManager.settings.glucoseTargetRangeSchedule?.unit ?? dataManager.loopManager.glucoseStore.preferredUnit ?? HKUnit.milligramsPerDeciliter
 
-                show(scheduleVC, sender: sender)
+                let editor = CorrectionRangeOverridesEditor(
+                    value: CorrectionRangeOverrides(
+                        preMeal: dataManager.loopManager.settings.preMealTargetRange,
+                        workout: dataManager.loopManager.settings.legacyWorkoutTargetRange,
+                        unit: unit
+                    ),
+                    unit: unit,
+                    minValue: dataManager.loopManager.settings.suspendThreshold?.quantity,
+                    onSave: { [dataManager] overrides in
+                        dataManager?.loopManager.settings.preMealTargetRange = overrides.preMeal?.doubleRange(for: unit)
+                        dataManager?.loopManager.settings.legacyWorkoutTargetRange = overrides.workout?.doubleRange(for: unit)
+                    }
+                )
+
+                let hostingController = ExplicitlyDismissibleModal(rootView: editor, onDisappear: {
+                    tableView.deselectRow(at: indexPath, animated: true)
+                })
+
+                self.present(hostingController, animated: true)
             case .suspendThreshold:
                 func presentSuspendThresholdEditor(initialValue: HKQuantity?, unit: HKUnit) {
                     let editor = SuspendThresholdEditor(
                         value: initialValue,
                         unit: unit,
+                        maxValue: dataManager.loopManager.settings.glucoseTargetRangeSchedule?.minLowerBound(),
```

Comment:
This seems strange. `maxValue` is set to the `minLowerBound`?

Reply:
By design; it doesn't make sense for the quantity below which Loop suspends all insulin delivery to be able to be higher than user's intended glucose target at any point in time